### PR TITLE
CI bug fix - unlink 2to3 before installing python3

### PR DESCRIPTION
**Pull request by candyapplecorn**
See the discord thread here: https://discord.com/channels/314987288398659595/401594186983407616/1196451948413595778

> the builds are failing because of devops stuff.
>I looked into the Mac build and it’s failing because Python is failing to symlink 2to3 since there are multiple Python versions being installed by homebrew and the Catalina image GitHub actions uses 
>Idk why the windows 10 build failed

## Related Ticket(s)

Related to #4982  because i found the issue there

## Short roundup of the initial problem
https://github.com/pypa/pipenv/issues/3831

## What will change with this Pull Request?
github's 2to3 will be unlinked so that the new 2to3 can be linked

### CODE REVIEW

File: .github/workflows/desktop-build.yml [hunk 1]
```diff
@@ -236,6 +236,11 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4
 
+      # RE: https://discord.com/channels/314987288398659595/401594186983407616/1196451948413595778
+      - name: Unlink 2to3
+        if: matrix.target == '10.15_Catalina'
+        run: brew unlink /usr/local/bin/2to3
```
Comment (tooomm) [inline .github/workflows/desktop-build.yml:242]: The log is stating to remove it. Did you try that?

```suggestion
        run: rm '/usr/local/bin/2to3'
```

**Comment by tooomm** [comment]
I tried to remove stuff as suggested in the logs, but it only leads to more and more other locations that would actually need to get removed. 
That's probably not the right solution to the problem.

@candyapplecorn any other idea to follow up here?

<br>

>the builds are failing because of devops stuff.
I looked into the Mac build and it’s failing because Python is failing to symlink 2to3 since there are multiple Python versions being installed by homebrew **and the Catalina image GitHub actions uses**

The GitHub Action uses a macOS 11 runner (Big Sur) by the way, just that we use an old Xcode SDK to get a build which is targeting 10.15 (Catalina) at the end.

**Comment by tooomm** [comment]
I realized that the issue is solved in the meantime and macOS builds do complete. No tweaks needed from our side I guess.
Windows is also working with my recent change.

Updated your other PR - it builds now as well.

**Comment by candyapplecorn** [comment]
thank you @tooomm

**Comment by tooomm** [comment]
As @njelich just pointed out in your first PR (4985) that the linking issue magically reappeared now without changes on our end.

Opened #5007

**Comment by tooomm** [comment]
Updated the PR with all "unlinks" as a workaround.
It works and the `homebrew` step finishes.

But now, the follow up Qt installation fails...

**Comment by njelich** [comment]
From the error message:

This environment is externally managed
╰─> To install Python packages system-wide, try brew install
    xyz, where xyz is the package you are trying to
    install.
    
    If you wish to install a non-brew-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip.

**Comment by njelich** [comment]
Upstream issue:

https://github.com/actions/runner-images/issues/4020

**Comment by ZeldaZach** [comment]
Superseded by 5033

**Pull request closed without merging**